File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/card/GetSetUpIntentUseCase.java
package com.lattis.ellipse.domain.interactor.card;

import com.lattis.ellipse.data.network.model.response.card.SetUpIntentResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.CardRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

public class GetSetUpIntentUseCase extends UseCase<SetUpIntentResponse> {

    private CardRepository cardRepository;

    @Inject
    public GetSetUpIntentUseCase(ThreadExecutor threadExecutor,
                          PostExecutionThread postExecutionThread,
                          CardRepository cardRepository) {
        super(threadExecutor, postExecutionThread);
        this.cardRepository = cardRepository;
    }

    @Override
    protected Observable<SetUpIntentResponse> buildUseCaseObservable() {
        return cardRepository.getSetUpIntent();
    }

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/card/GetSetUpIntentUseCase.kt
package com.lattis.domain.usecase.card

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.models.SetUpIntent
import com.lattis.domain.repository.CardRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class GetSetUpIntentUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val cardRepository: CardRepository
) : UseCase<SetUpIntent>(threadExecutor, postExecutionThread) {
    override fun buildUseCaseObservable(): Observable<SetUpIntent> {
        return cardRepository.getSetUpIntent()
    }

}